app/experimenter/jetstream/client.py

```diff
@@ -57,12 +59,32 @@ def get_results_metrics_map(data, primary_outcome_slugs, secondary_outcome_slugs
         )
     )
 
+    if outcomes_metadata is not None:
+        BYPASS_JETSTREAM_CHECK = False
+        metrics_set_from_jetstream = set(
+            chain.from_iterable(
+                [
+                    outcomes_metadata[slug]["metrics"]
+                    + outcomes_metadata[slug]["default_metrics"]
+                    for slug in outcomes_metadata
+                ]
+            )
+        )
+    else:
+        BYPASS_JETSTREAM_CHECK = True
+
     for metric in primary_outcome_metrics:
-        RESULTS_METRICS_MAP[metric.slug] = set([Statistic.BINOMIAL])
-        primary_metrics_set.add(metric.slug)
+        # validate against jetstream metadata unless we couldn't get it
+        if BYPASS_JETSTREAM_CHECK or metric.slug in metrics_set_from_jetstream:
+            RESULTS_METRICS_MAP[metric.slug] = set(
+                [Statistic.BINOMIAL, Statistic.MEAN, Statistic.COUNT, Statistic.PERCENT]
```

Comment:
If these two lists are the same, and static, can we move it up to the top and make it a constant?